rewrite-maven/src/test/java/org/openrewrite/maven/RemoveRedundantDependencyVersionsTest.java

```diff
@@ -1530,4 +1530,50 @@ void keepVersionIfManagedIsNonExact() {
           )
         );
     }
+
+    @Test
+    @Issue("https://github.com/openrewrite/rewrite/discussions/4386")
+    void dontOverrideDependencyConfigurations() {
+        rewriteRun(
+          spec -> spec.recipe(new RemoveRedundantDependencyVersions(null, null, RemoveRedundantDependencyVersions.Comparator.GTE, null)),
+          pomXml(
+            """
+                <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+                    <modelVersion>4.0.0</modelVersion>
+                    <parent>
+                        <groupId>org.springframework.boot</groupId>
+                        <artifactId>spring-boot-starter-parent</artifactId>
+                        <version>2.3.12.RELEASE</version>
+                    </parent>
+                    <groupId>com.example</groupId>
+                    <artifactId>acme</artifactId>
+                    <version>0.0.1-SNAPSHOT</version>
+                    <dependencyManagement>
+                        <dependencies>
+                            <dependency>
+                                <groupId>org.springframework.boot</groupId>
+                                <artifactId>spring-boot-starter-web</artifactId>
+                                <!-- The version number should be dropped -->
+                                <version>2.2.13.RELEASE</version>
```

Comment:
I was thinking remove the version, but leave exclusions, scope, etc.  My colleagues agree, but I wouldn't say we've given it a ton of thought.  There's no clear-cut right answer in my opinion.  I can see an argument for making this configurable.  But leaving the managed version is likely more problematic than removing it altogether (current behavior).

Reply:
I'd remove the managed dep if completely redundant, and it met the conditions (GTE, LTE, etc).  I guess you could test for equality, but I'd probably just remove if met the precondition and only had GAV coordinates.